### PR TITLE
Episode component: podcasts, episodes, comments

### PR DESCRIPTION
**Pull request by holly-christensen**
- podcast component with no reviews and a nonfunctional follow button
- podcast component has list of episodes with paginated requests
- episode component has episode details
- episode component has a list of comments
- episode component has "write a comment" inputs in SecureContent, as well as comment like and dislike buttons
- episode component shows the podcast creator (if exists) and links to their profile page

### CODE REVIEW

**Comment by holly-christensen** [comment]
things that need work: 
- the comment like/dislike icons **do not** turn a different color to indicate that the comment been liked/disliked by the current user
- we are missing some things like "language", and other random metadata for the episode but it doesn't feel very important right now

--> please let me know if there is anything else that should be on the page that we are missing

File: app/schemas/creators-schema.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 const creatorsSchema = mongoose.Schema({
-        userId: {
-                type: mongoose.Schema.Types.ObjectId,
-                ref: 'User'
-        },
+        // userId: {
+        //         type: mongoose.Schema.Types.ObjectId,
+        //         ref: 'User'
+        // },
+        userId: String,
```
Comment (MMorehead) [inline app/schemas/creators-schema.js:8]: Should this type be the mongoose reference instead of String?

File: app/schemas/podcasts-schema.js [hunk 2]
```diff
@@ -5,13 +5,19 @@ import creatorsSchema from "./creators-schema.js";
 
 const podcastsSchema = mongoose.Schema({
         podcastId: String,
-        creator: {creatorsSchema},
+        creator: {
```
Comment (MMorehead) [inline app/schemas/podcasts-schema.js:8]: maybe name this creatorId just for consistency?

File: podcast-app/src/actions/user-actions.js [hunk 3]
```diff
@@ -1,37 +1,47 @@
...
-            password: userDetails.password,
-        },
-        type: USER_CONSUMER,
-        following: [],
-        comments: [],
-        reviews: []
-    }
-    const user = await service.createUser(newUser);
-    console.log(user)
+export const signUpUser = async (dispatch, userDetails) => {
```
Comment (MMorehead) [inline podcast-app/src/actions/user-actions.js:36]: Thank you for moving this into actions!

File: podcast-app/src/components/EpisodeDetails.js [hunk 4]
```diff
@@ -1,11 +1,242 @@
...
+        }
+
+        return (
+            <>
+                <div className={"container col-9 mt-5"}>
+                    <h1>{episodeDetails.title}</h1>
+                    <Link to={`/profile/${creatorDetails.userId}`} className={"text-decoration-none mb-4"}>
+                        <h6>{creatorDetails.username}</h6>
+                    </Link>
+                    <p>{JSON.stringify(profile)}</p>
```
Comment (MMorehead) [inline podcast-app/src/components/EpisodeDetails.js:173]: Can we remove this? I assume it was just here for debugging

File: podcast-app/src/components/EpisodeDetails.js [hunk 5]
```diff
@@ -1,11 +1,242 @@
...
+            setCreatorDetails({...creatorDetails, username: podcastCreator.username, userId: podcastCreator.userId})
+            return podcastCreator;
+        }
+
+        return (
+            <>
+                <div className={"container col-9 mt-5"}>
+                    <h1>{episodeDetails.title}</h1>
+                    <Link to={`/profile/${creatorDetails.userId}`} className={"text-decoration-none mb-4"}>
+                        <h6>{creatorDetails.username}</h6>
```
Comment (MMorehead) [inline podcast-app/src/components/EpisodeDetails.js:171]: could you do a null check here just so if the username is null it won't display null

**Review by MMorehead — APPROVED** [review]
Looks great!

File: podcast-app/src/components/EpisodeDetails.js (hunk 4 above)
Comment (holly-christensen) [reply podcast-app/src/components/EpisodeDetails.js:173]: yep

File: app/schemas/creators-schema.js (hunk 1 above)
Comment (holly-christensen) [reply app/schemas/creators-schema.js:8]: Technically, yes. But I was having time-consuming issues with it and I felt like getting it to work was more important at this point

File: podcast-app/src/components/EpisodeDetails.js (hunk 5 above)
Comment (holly-christensen) [reply podcast-app/src/components/EpisodeDetails.js]: fixed

File: app/schemas/creators-schema.js (hunk 1 above)
Comment (MMorehead) [reply app/schemas/creators-schema.js:8]: that's okay with me then

**Pull request merged**